images/fcos-bastion-image/root/usr/bin/prune_nodes

```diff
@@ -0,0 +1,202 @@
+  fi
+  
+  pushd /opt/tftpboot || echo "<3>pushd to /opt/tftpboot failed"
+  rm -fr "$CLUSTER"
+  if [[ ${#grub_macs[@]} -gt 0 ]]; then
+    for mac in "${grub_macs[@]}"; do
+      grubcfg="grub.cfg-01-$(tr ':' '-' <<< "${mac}")"
+      rm -f "${grubcfg}"
+    done
+  fi
+  popd || exit 1
+}
+
+function reboot_stuck_pdu() {
+  while read -r nodeh nodeu nodep pdu_uri; do
+    if [[ -n ${nodeh} && -n ${nodeu} && -n ${nodep} ]]; then
+      if [[ -n ${pdu_uri} ]] && ! ipmitool -I lanplus -H "${nodeh}" -U "${nodeu}" -P "${nodep}" \
+        power status | grep -Eq ' on| off'; then
+        pdu_host=${pdu_uri%%/*}
+        pdu_host=${pdu_host##*@}
+        pdu_socket=${pdu_uri##*/}
+        pdu_creds=${pdu_uri%%@*}
+        pdu_user=${pdu_creds%%:*}
+        pdu_pass=${pdu_creds##*:}
+        echo "${pdu_pass}" > /tmp/ssh-pass
+        timeout -s 9 10m sshpass -f /tmp/ssh-pass ssh "${pdu_user}@${pdu_host}" <<EOF
+olReboot $pdu_socket
+quit
+EOF
+        echo "<3>${nodeh} - pdu reboot"
+        pdu_reboot=true
+      fi
+    fi
+  done
+}
+
+function wait_for_node_status() {
+ plan="$1"
+ status="$2"
+  while read -r nodeh nodeu nodep pdu_uri; do
+    if [[ -n ${nodeh} && -n ${nodeu} && -n ${nodep} ]]; then
+      echo "<4>Waiting for 2 min before checking..." && sleep 120
+      retry_max=32
+      while [ $retry_max -gt 0 ] && ! ipmitool -I lanplus -H "${nodeh}" -U "${nodeu}" -P "${nodep}" power status | \
+        grep -Eq "${status}"; do
+        echo "<4>Waiting for ${nodeh} ${plan}..."
+        sleep 15
+        retry_max=$(( retry_max - 1 ))
+      done
+      [ $retry_max -le 0 ] && echo "<3>${nodeh} needs investigation - ${plan}"
+    else
+      echo "<3>Invalid line: check the node variables"
+    fi
+  done
+}
+
+function wipe_disk() {
+  while read -r nodeh nodeu nodep pdu_uri; do
+    ipmitool -I lanplus -H "${nodeh}" -U "${nodeu}" -P "${nodep}" \
+      chassis bootparam set bootflag force_pxe options=PEF,watchdog,reset,power
+    ipmitool -I lanplus -H "${nodeh}" -U "${nodeu}" -P "${nodep}" power reset
+  done
+}
+ 
+
+############
+ ## Main ##
+############
+
+if [ "$#" -ne 1 ]; then
+    echo "<3>Usage: $0 <cluster-name>"
+    return 1
+fi
+
+CLUSTER=${1}
+INTERNAL_NET=${INTERNAL_NET:-192.168.90.0/24}
+DOCKER_NET=${DOCKER_NET:-172.17.0.0/16}
+reserved_hosts="/etc/hosts_pool_reserved"
+var_dir="/var/builds/${CLUSTER}"
+conf_files=(/opt/dhcpd/root/etc/dnsmasq.conf
+            /opt/bind9_zones/zone
+            /opt/bind9_zones/internal_zone.rev
+            /opt/bind9_zones/external_zone.rev
+            /etc/vips_reserved
+            "${reserved_hosts}")
+
+readarray -t nodes < <(awk -F, -v cluster="${CLUSTER}" '$0 ~ cluster {print $12,$15,$16,$23}' "${reserved_hosts}")
+readarray -t node_ips < <(awk -F, -v cluster="${CLUSTER}" '$0 ~ cluster {print $2}' "${reserved_hosts}")
+readarray -t grub_macs < <(awk -F, -v cluster="${CLUSTER}" '$0 ~ cluster {print $1}' "${reserved_hosts}")
```

Comment:
there is a strong dependency on the number of the column here. Can we avoid it similarly to how we do in other code (see the scripts in openshift/release). The idea is that we should minimize the amount of changes required if we change the columns of the csv.

Reply:
done